File: tools/create-connector/templates/typescript/handlers.ts
/**
 * Your logic goes here.
 *
 * Nothing in this file imports the Nimbus SDK or the MCP SDK, and that is deliberate: business
 * logic you can test without a wire protocol is business logic you will actually test. `main.ts`
 * is the only file that knows a protocol exists.
 */

export interface EchoInput {
  readonly text: string;
}

export async function echo(input: EchoInput): Promise<{ readonly text: string }> {
  return { text: input.text };
}

File: tools/create-connector/templates/python/src/nimbus_quickstart_connector/handlers.py
"""Your logic goes here.

Nothing in this file imports the Nimbus SDK or the MCP package, deliberately: logic you
can test without a wire protocol is logic you will actually test. ``main.py`` is the
only file that knows a protocol exists.
"""

from __future__ import annotations


def echo(text: str) -> dict[str, str]:
    return {"text": text}
